AirGradient-DIY.ino

```diff
@@ -1,117 +1,161 @@
+
+const bool hasPM = true;
+const bool hasCO2 = true;
+const bool hasSHT = true;
 
-// set sensors that you do not use to false
-boolean hasPM=true;
-boolean hasCO2=true;
-boolean hasSHT=true;
+const char* ssid = "PleaseChangeMe";
+const char* password = "PleaseChangeMe";
 
-// set to true if you want to connect to wifi. The display will show values only when the sensor has wifi connection
-boolean connectWIFI=true;
+const int8_t port = 9926;
 
-// change if you want to send the data to another server
-String APIROOT = "http://geerli.net:9926/";
+long lastUpdate;
+//The frequency of measurement updates
+const int8_t updateFrequency = 5000; 
 
-void setup(){
+int counter = 0;
+
+// Config End
+//
+SSD1306Wire display(0x3c, SDA, SCL);
+ESP8266WebServer server(port);
+//
+
+void setup() {
   Serial.begin(9600);
 
+  // Init Display.
   display.init();
   display.flipScreenVertically();
   showTextRectangle("Init", String(ESP.getChipId(),HEX),true);
 
+  // Enable enabled sensors.
   if (hasPM) ag.PMS_Init();
   if (hasCO2) ag.CO2_Init();
   if (hasSHT) ag.TMP_RH_Init(0x44);
 
-  if (connectWIFI) {
-    showTextRectangle("Connect", "to WiFi",true);    
-    connectToWifi();
+  // Setup and wait for wifi.
+  WiFi.begin(ssid, password);
+  Serial.println("");
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
   }
-  delay(2000);
+
+  Serial.println("");
+  Serial.print("Connected to ");
+  Serial.println(ssid);
+  Serial.print("IP address: ");
+  Serial.println(WiFi.localIP());
+  Serial.print("MAC address: ");
+  Serial.println(WiFi.macAddress());
+
+  server.on("/", HandleRoot);
+  server.on("/metrics", HandleRoot);
+  server.onNotFound(HandleNotFound);
+
+  server.begin();
+  Serial.println("HTTP server started at ip " + WiFi.localIP().toString() );
+  showTextRectangle("Listening To", WiFi.localIP().toString() ,true);
 }
 
-void loop(){
+void loop() {
+  long t = millis();
+
+  server.handleClient();
+  updateScreen(t);
+}
 
-  // create payload
+String GenerateMetrics() {
+  String message = "";
 
-  String payload = "{\"wifi\":" + String(WiFi.RSSI()) + ",";
+  String idString = "id=\"" + String(deviceId) + "\" ";
 
   if (hasPM) {
-    int PM2 = ag.getPM2_Raw();
-    payload=payload+"\"pm02\":" + String(PM2);
-    showTextRectangle("PM2",String(PM2),false);
-    delay(3000);
+    int stat = ag.getPM2_Raw();
+
+    message += "# HELP pm02 Particulat Matter PM2.5 value\n";
+    message += "# TYPE pm02 gauge\n";
+    message += "pm02";
+    message += "{";
+    message += idString;
+    message += "\"} ";
+    message += String(stat);
+    message += "\n";
   }
 
   if (hasCO2) {
-    if (hasPM) payload=payload+",";
-    int CO2 = ag.getCO2_Raw();
-    payload=payload+"\"rco2\":" + String(CO2);
-    showTextRectangle("CO2",String(CO2),false);
-    delay(3000);
+    int stat = ag.getCO2_Raw();
+
+    message += "# HELP rc02 CO2 value, in ppm\n";
+    message += "# TYPE rc02 gauge\n";
+    message += "pm02";
```

Comment:
This should produce the metric `rc02` and not the one from the previous block.

Reply:
Whoops, surprised we didn't catch that last night. Then again, we had been up for a loonngg time.